src/algebra/homology/homotopy.lean

```diff
@@ -270,6 +282,158 @@ def comp_right_id {f : C ⟶ C} (h : homotopy f (𝟙 C)) (g : C ⟶ D) : homoto
+these morphisms are homotopic to `0` and provide some convenient simplification
+lemmas that give a degreewise description of `hd+dh`, depending on whether we have 
+two differentials going to and from a certain degree, only one, or none.
+-/
+
+/-- The null homotopic map associated to a family `hom` of morphisms `C_i ⟶ D_j`.
+This is the same datum as for the field `hom` in the structure `homotopy`. For
+this definition, we do not need the field `zero` of that structure 
+as this definition uses only the maps `C_i ⟶ C_j` when `c.rel j i`. -/
+def null_homotopic_map (hom : Π i j, C.X i ⟶ D.X j) : C ⟶ D :=
+{ f      := λ i, d_next i hom + prev_d i hom,
+  comm'  := λ i j hij,
+  begin
+    have eq1 : prev_d i hom ≫ D.d i j = 0,
+    { rcases h : c.prev i with _|⟨i',w⟩,
+      { dsimp [prev_d], rw h, erw zero_comp, },
+      { rw [prev_d_eq hom w, category.assoc, D.d_comp_d' i' i j w hij, comp_zero], }, },
+    have eq2 : C.d i j ≫ d_next j hom = 0,
+    { rcases h : c.next j with _|⟨j',w⟩,
+      { dsimp [d_next], rw h, erw comp_zero, },
+      { rw [d_next_eq hom w, ← category.assoc, C.d_comp_d' i j j' hij w, zero_comp], }, },
+    rw [d_next_eq hom hij, prev_d_eq hom hij, preadditive.comp_add, preadditive.add_comp,
+      eq1, eq2, add_zero, zero_add, category.assoc], 
+  end }
+
+/-- Variant of `null_homotopic_map` where the input consists only of the
+relevant maps `C_i ⟶ D_j` such that `c.rel j i`. -/
+def null_homotopic_map' (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) : C ⟶ D :=
+null_homotopic_map (λ i j, dite (c.rel j i) (h i j) (λ _, 0))
+
+/-- Tautological construction of the `homotopy` to zero for maps constructed by
+`null_homotopic_map`, at least when we have the `zero'` condition. -/
+@[simps]
+def null_homotopy (hom : Π i j, C.X i ⟶ D.X j) (zero' : ∀ i j, ¬ c.rel j i → hom i j = 0) :
+  homotopy (null_homotopic_map hom) 0 :=
+{ hom := hom,
+  zero' := zero',
+  comm := by { intro i, rw [homological_complex.zero_f_apply, add_zero], refl, }, }
+
+/-- Homotopy to zero for maps constructed with `null_homotopic_map'` -/
+@[simps]
+def null_homotopy' (h : Π i j, c.rel j i → (C.X i ⟶ D.X j)) :
+  homotopy (null_homotopic_map' h) 0 :=
+begin
+  apply null_homotopy (λ i j, dite (c.rel j i) (h i j) (λ _, 0)),
+  intros i j hij,
+  dsimp,
+  rw [dite_eq_right_iff],
+  intro hij',
+  exfalso,
+  exact hij hij',
+end
+
+/-! This lemma and the following ones can be used in order to compute
+the degreewise morphisms induced by the null homotopic maps constructed
+with `null_homotopic_map` or `null_homotopic_map'` -/
+@[simp]
+lemma null_homotopic_map_f {k₂ k₁ k₀ : ι} (r₂₁ : c.rel k₂ k₁) (r₁₀ : c.rel k₁ k₀)
+  (hom : Π i j, C.X i ⟶ D.X j) :
+  (null_homotopic_map hom).f k₁ = C.d k₁ k₀ ≫ hom k₀ k₁ + hom k₁ k₂ ≫ D.d k₂ k₁ :=
+by { dsimp [null_homotopic_map], rw [d_next_eq hom r₁₀, prev_d_eq hom r₂₁], }
+
+@[simp]
+lemma null_homotopic_map'_f {k₂ k₁ k₀  : ι} (r₂₁ : c.rel k₂ k₁) (r₁₀ : c.rel k₁ k₀)
```

Comment:
Here, the `'` was placed in the middle of the name, which makes sense to me. But in the other names the `'` is at the end. Please make this consistent.

(I would vote for `'` in the middle of the name.)

Reply:
Ok, I have put the apostrophes in the middle of the names.